internal/apis/certmanager/validation/certificaterequest.go

```diff
@@ -113,13 +113,24 @@ func validateCertificateRequestSpecRequest(crSpec *cmapi.CertificateRequestSpec,
 		pki.CertificateTemplateValidateAndOverrideKeyUsages(keyUsage, extKeyUsage),
 	)
 	if err != nil {
-		el = append(el, field.Invalid(fldPath.Child("request"), crSpec.Request, err.Error()))
+		// truncate the request to avoid creating a ridiculously long error message with the whole CSR in it
+		el = append(el, field.Invalid(fldPath.Child("request"), truncateString(string(crSpec.Request)), err.Error()))
 		return el
 	}
 
 	return el
 }
 
+func truncateString(s string) string {
+	const maxLength = 100
+
+	if len(s) <= maxLength {
+		return s
+	}
+
+	return s[:maxLength]
```

Comment:
```suggestion
	return s[:maxLength] + "..."
```

To indicate that the value is actually truncated?